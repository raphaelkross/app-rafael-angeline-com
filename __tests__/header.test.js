/* eslint-env jest */

import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Header from "../components/header";

describe("Header", () => {
	it("should match complete layout snapshot", () => {
		const items = [
			{ href: "#1", label: "Home" },
			{
				href: "#2",
				label: "About",
				children: [
					{ href: "#sub", label: "Sub" },
					{ href: "#sub2", label: "Sub2" }
				]
			}
		];

		const component = renderer.create(<Header items={items} />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("should still render even without menu items", () => {
		const items = [];
		let header = shallow(<Header items={items} />);

		expect(header.find("header").length).toEqual(1);

		header = shallow(<Header />);

		expect(header.find("header").length).toEqual(1);
	});

	it("should render right amout of links", () => {
		const items = [
			{ href: "#1", label: "Home" },
			{
				href: "#2",
				label: "About",
				children: [
					{ href: "#sub", label: "Sub" },
					{ href: "#sub2", label: "Sub2" }
				]
			}
		];

		let header = shallow(<Header items={items} />);

		const menu = header.find("nav");

		// Render 4 links.
		expect(menu.find("a").length).toEqual(4);
	});

	it("should render first-level items properly", () => {
		const items = [
			{ href: "#1", label: "Home" },
			{
				href: "#2",
				label: "About",
				children: [
					{ href: "#sub", label: "Sub" },
					{ href: "#sub2", label: "Sub2" }
				]
			}
		];

		let header = shallow(<Header items={items} />);

		// First item should be a link without children.
		const itemOne = header.find("nav > ul > li").at(0);

		expect(itemOne.find("a").length).toEqual(1);

		const itemOneLink = itemOne.find("a").at(0);

		expect(itemOneLink.prop("href")).toEqual("#1");
		expect(itemOneLink.children("span").prop("children")).toEqual("Home");
	});

	it("should render items w/ children properly", () => {
		const items = [
			{ href: "#1", label: "Home" },
			{
				href: "#2",
				label: "About",
				children: [
					{ href: "#sub", label: "Sub" },
					{ href: "#sub2", label: "Sub2" }
				]
			}
		];

		let header = shallow(<Header items={items} />);

		// Check the amount of links being rendered.
		const itemTwo = header.find("nav > ul > li").at(1);

		expect(itemTwo.find("a").length).toEqual(3);

		// Must have sub-menu.
		expect(itemTwo.find("ul").length).toEqual(1);
		expect(itemTwo.find("ul > li").length).toEqual(2);

		// Rendering main link properly?
		const itemTwoLink = itemTwo.children("a").at(0);

		expect(itemTwoLink.prop("href")).toEqual("#2");
		expect(itemTwoLink.children("span").prop("children")).toEqual("About");

		// Rendering Submenu links properly.
		const ItemSubMenu = itemTwo.find("ul > li a").at(0);

		expect(ItemSubMenu.prop("href")).toEqual("#sub");
		expect(ItemSubMenu.children("span").prop("children")).toEqual("Sub");
	});
});
