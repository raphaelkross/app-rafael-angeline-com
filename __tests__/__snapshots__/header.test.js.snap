// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match complete layout snapshot 1`] = `
<header
  className="jsx-1373952718"
>
  <div
    className="jsx-1373952718 container"
  >
    <nav
      className="jsx-1373952718"
    >
      <ul>
        <li>
          <a
            href="#1"
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li>
          <a
            href="#2"
          >
            <span>
              About
            </span>
          </a>
          <ul>
            <li>
              <a
                href="#sub"
              >
                <span>
                  Sub
                </span>
              </a>
            </li>
            <li>
              <a
                href="#sub2"
              >
                <span>
                  Sub2
                </span>
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </nav>
    <div
      className="jsx-1373952718 responsive-trigger"
    >
      <a
        className="jsx-1373952718 responsive-trigger-link"
        href="#"
        onClick={[Function]}
      >
        <svg
          className="jsx-1373952718 feather feather-menu"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            className="jsx-1373952718"
            x1="3"
            x2="21"
            y1="12"
            y2="12"
          />
          <line
            className="jsx-1373952718"
            x1="3"
            x2="21"
            y1="6"
            y2="6"
          />
          <line
            className="jsx-1373952718"
            x1="3"
            x2="21"
            y1="18"
            y2="18"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    className="jsx-1373952718 modal"
  >
    <a
      className="jsx-1373952718 modal-close"
      href="#"
      onClick={[Function]}
    >
      <svg
        className="jsx-1373952718 feather feather-x"
        fill="none"
        height="24"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          className="jsx-1373952718"
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          className="jsx-1373952718"
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </a>
    <div
      className="jsx-1373952718"
    >
      <ul>
        <li>
          <a
            href="#1"
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li>
          <a
            href="#2"
          >
            <span>
              About
            </span>
          </a>
          <ul>
            <li>
              <a
                href="#sub"
              >
                <span>
                  Sub
                </span>
              </a>
            </li>
            <li>
              <a
                href="#sub2"
              >
                <span>
                  Sub2
                </span>
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</header>
`;
